Readme updated + Examples for css added (#2)
kontent-ai/integration-template-react: README.md

[![Discord][discord-shield]][discord-url]

# Kontent.ai React Integration Template

> This template can be used to jumpstart your own integration development with Kontent.ai. It contains all the neccessary tools for creating a new [Custom Element](https://kontent.ai/learn/tutorials/develop-apps/integrate/content-editing-extensions/), a UI extension for content editors. 

Reviewer: ```suggestion
This template can be used to jumpstart your own custom element development with Kontent.ai. It contains all the necessary tools for creating a new [Custom Element](https://kontent.ai/learn/tutorials/develop-apps/integrate/content-editing-extensions/), a UI extension for content editors. 
```

You can inspire yourself by browsing already created integrations [**here**](https://github.io/topics/kontent-ai-integration).

If you wish to include your integration into the mentioned list, please add the **kontent-ai-integration** topic into your github integration repository. 

> Additional information and tutorials can be found on [kontent-ai.github.io](https://kontent-ai.github.io/).

Reviewer: ```suggestion
Additional Kontent.ai GitHub resources and tutorials can be found on [kontent-ai.github.io](https://kontent-ai.github.io/).
```

<br />

## Getting Started

> The integration is created with [Create React App](https://create-react-app.dev/). 

Reviewer: ```suggestion
The integration is created with [Create React App](https://create-react-app.dev/). 

This template aims to showcase the possibilities. It is storing the data from the displayed input, it is observing another text element specified in the configuration, and displays the context provided by the Kontent.ai app.

It is also showcasing how you can select other content items as well as assets within the custom element.
```

[![Discord][discord-shield]][discord-url]

# Kontent.ai React Integration Template

This template can be used to jumpstart your own integration development with Kontent.ai. It contains all the neccessary tools for creating a new [Custom Element](https://kontent.ai/learn/tutorials/develop-apps/integrate/content-editing-extensions/), a UI extension for content editors. 

> 

Reviewer: Can we add the features? 

* Parent frame check
* Height adjustments
* OnDisabled handling
* Watched wlement
* Input element
* Selected assets
* Selected Items

> 

Reviewer: ```suggestion
```

You can inspire yourself by browsing already created integrations [**here**](https://github.io/topics/kontent-ai-integration).

If you wish to include your integration into the mentioned list, please add the **kontent-ai-integration** topic into your github integration repository. 

> Additional information and tutorials can be found on [kontent-ai.github.io](https://kontent-ai.github.io/).

Reviewer: Great - we might introduce another badge for thet - what do you thing.

And we need to sync about the purpose of the kontent-ai.github.io

### Initialize the custom element

Before you start any interaction with the Kontent.ai app, you'll need **initialize the custom element**. You can do that by calling the `init` function from the provided API like so https://github.com/kontent-ai/custom-element-template-react/blob/e184179039aa705a82722d778e503dfb511f7115/src/IntegrationApp.tsx#L18-L29
and waiting until the callback you passed into the `init` function is called. You'll receive the custom element configuration object of your specification, as well as additional context - more on that in [the documentation](https://kontent.ai/learn/reference/custom-elements-js-api/#a-init-method). 
You can leverage state in React to store the important configuration while setting it inside of the initialization callback. 

```ts
const [config, setConfig] = useState<Config | null>(null);

useEffect(() => {
  CustomElement.init((element, context) => {
    if (!isValidConfig(element.config)) {
      throw new Error('Not the config this element expects');
    }
    setConfig(element.config);
    // more logic
  });
}, []);
```

### Disable your element if needed

> In certain circumstances, the Kontent.ai app might instruct your custom element to display itself in a **disabled state**. Disabled state means that the value of your element cannot be changed. This can happen when the edited item is published (therefore cannot be edited), or the current user does not have permission to edit the custom element.

Reviewer: the current user does not have permission to edit the custom element. -> this information is not in Custom element API - or is it?

Reviewer: You are correct, it is -> https://kontent.ai/learn/reference/custom-elements-js-api/#a-ondisabledchanged-method

This value is accessible through the Delivery API, and can be used and rendered on your website, or in your application.

### Adjust the height

In some cases, the default height of custom elements will not be enough for the application you are trying to render inside of it. In that case, you can set your custom height with the [`CustomElement.setHeight`](https://kontent.ai/learn/reference/custom-elements-js-api/#a-setheight-method) function. This resize function can be  changed as the required height value itself changes, which helps you omit the unpleasant scrolling of the iframe your element is rendered in. 
In the following React example we update the element's size whenever some value affecting the element's size changes and set the height to the document's height with a minimum of 100px.

```ts
const updateSize = useCallback(() => {
  const newSize = Math.max(document.documentElement.offsetHeight, 100);

  CustomElement.setHeight(Math.ceil(newSize));
}, []);

useLayoutEffect(() => {
  updateSize();
}, [updateSize, currentValue, searchResults]);
```

### More functions

> You can find much all of the available Custom Element functions in our [Custom Element API reference](https://kontent.ai/learn/reference/custom-elements-js-api/).

Reviewer: ```suggestion
You can find all of the available Custom Element functions in our [Custom Element API reference](https://kontent.ai/learn/reference/custom-elements-js-api/).
```
